### PR TITLE
Add metadata interfaces for lake storage

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [x] others

### CODE REVIEW

**Review by imay — DISMISSED** [review]

File: be/src/storage/lake/tablet.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Status Tablet::delete_metadata(int64_t version) {
+    return _mgr->delete_tablet_metadata(_group, _id, version);
+}
+
+Status Tablet::delete_metadata() {
+    return Status::NotSupported("Tablet::delete_metadata");
+}
+
+Status Tablet::put_txnlog(const TxnLog& log) {
```
Comment (wyb) [inline be/src/storage/lake/tablet.cpp:32]:
```suggestion
Status Tablet::put_txn_log(const TxnLog& log) {
```

File: be/src/storage/lake/tablet_manager.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+private:
+    Status put_tablet_metadata(const std::string& group, const TabletMetadata& metadata);
+    StatusOr<TabletMetadata> get_tablet_metadata(const std::string& group, int64_t tablet_id, int64_t version);
+    Status delete_tablet_metadata(const std::string& group, int64_t tablet_id, int64_t version);
+    StatusOr<MetadataIterator> list_tablet_metadata(const std::string& group);
+    StatusOr<MetadataIterator> list_tablet_metadata(const std::string& group, int64_t tablet_id);
+
+    StatusOr<TxnLog> get_txnlog(const std::string& group, int64_t tablet_id, int64_t txn_id);
+    Status put_txnlog(const std::string& group, int64_t tablet_id, const TxnLog& log);
```
Comment (wyb) [inline be/src/storage/lake/tablet_manager.h:48]:
```suggestion
    Status put_txnlog(const std::string& group, const TxnLog& log);
```

Comment (sduzh) [reply be/src/storage/lake/tablet_manager.h]: updated, PTAL

File: be/src/storage/lake/tablet.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/storage/lake/tablet.cpp]: Updated, PTAL

**Review by wyb — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**